MyXamarinApp.API/Controllers/EmployeeController.cs

```diff
@@ -21,21 +23,79 @@ public EmployeeController(IEmployeeRepository employeeRepository)
         public async Task<ActionResult<List<EmployeeModel>>> GetAllEmployees()
         {
             var employees = await _employeeRepository.GetAllEmployees();
-            return employees;
+            if (employees != null)
+            {
+                return Ok(employees);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<List<EmployeeModel>>> GetEmployeeById(Guid id)
+        {
+            var employees = await _employeeRepository.GetEmployeeById(id);
+            if (employees.Any())
+            {
+                return Ok(employees);
+            }
+            else
+            {
+                return NotFound();
+            }
         }
 
         [HttpPost]
         public async Task<ActionResult<EmployeeModel>> AddEmployee([FromBody] EmployeeModel employee)
         {
             var newEmployee = await _employeeRepository.AddEmployee(employee);
-            return newEmployee;
+            if(newEmployee != null)
+            {
+                return CreatedAtAction(nameof(AddEmployee), newEmployee.EmployeeGuid, newEmployee);
+            }
+            else
+            {
+                return UnprocessableEntity(employee);
+            }
         }
 
-        [HttpPost]
-        public async Task<ActionResult> RemoveEmployee([FromBody] int id)
+        [HttpPut]
+        public async Task<ActionResult<EmployeeModel>> UpdateEmployee([FromBody] EmployeeModel employee)
+        {
+            var currentEmployee = await _employeeRepository.GetEmployeeById(employee.EmployeeGuid);
+            if (currentEmployee.Any())
+            {
+                var updatedEmployee = await _employeeRepository.UpdateEmployee(employee);
+                if(updatedEmployee != null)
+                {
+                    return NoContent();
+                }
+                else
+                {
+                    return NotFound();
```

Comment:
If you try an update and the update failed, then should not be a 404, it could be a 422 or a 400